File: minigames-api/src/main/java/net/minevn/minigames/gadgets/Sword.java
package net.minevn.minigames.gadgets;

import org.bukkit.Material;
import org.bukkit.inventory.ItemFlag;
import org.bukkit.inventory.ItemStack;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Sword {
	private String id;
	private Material type;
	private short data;
	private String name;
	private String hitSound;
	private List<String> description;

	public Sword(String id, String name, Material type, short data, String hitSound, List<String> description) {
		this.id = id;
		this.type = type;
		this.data = data;
		this.name = name;
		this.hitSound = hitSound;
		this.description = description;
	}

	public Material getType() {
		return type;
	}

	public short getData() {
		return data;
	}

	public String getName() {
		return name;
	}

	public String getID() {
		return id;
	}

	public String getHitSound() {
		return hitSound;
	}

	public List<String> getDescription() {
		return description;
	}

	public ItemStack getItem() {
		var item = new ItemStack(type, 1, data);
		var im = item.getItemMeta();
		im.setDisplayName(name);
		im.setUnbreakable(true);
		im.addItemFlags(ItemFlag.HIDE_ENCHANTS, ItemFlag.HIDE_UNBREAKABLE, ItemFlag.HIDE_ATTRIBUTES);
		item.setItemMeta(im);
		return item;
	}

	public ItemStack replaceItem(ItemStack item) {
		var sname = item.getI18NDisplayName();
		item.setType(type);
		item.setDurability(data);
		var im = item.getItemMeta();
		im.setDisplayName(name + " §b(" + sname + ")");
		im.setUnbreakable(true);
		im.addItemFlags(ItemFlag.HIDE_UNBREAKABLE, ItemFlag.HIDE_ATTRIBUTES);
		item.setItemMeta(im);
		return item;
	}

	//region static
	private static Map<String, Sword> list = new HashMap<>();

	public static Sword get(String id) {
		return list.get(id);
	}

	public static void load() {
	}
	//endregion
}

File: minigames-plugin/src/main/java/net/minevn/minigames/gadgets/Sword.kt
package net.minevn.minigames.gadgets

import net.minevn.minigames.Configs
import net.minevn.minigames.Minigames
import net.minevn.minigames.colorCodes
import org.bukkit.Material
import org.bukkit.configuration.file.YamlConfiguration
import org.bukkit.inventory.ItemFlag
import org.bukkit.inventory.ItemStack
import java.io.File
import java.util.logging.Level

class Sword(val iD: String, val name: String, val type: Material, val data: Short,
            val hitSound: String?, val description: List<String>) {

	fun getItem(): ItemStack {
		val item = ItemStack(type, 1, data)
		val im = item.itemMeta
		im.setDisplayName(name)
		im.isUnbreakable = true
		im.addItemFlags(ItemFlag.HIDE_ENCHANTS, ItemFlag.HIDE_UNBREAKABLE, ItemFlag.HIDE_ATTRIBUTES)
		item.itemMeta = im
		return item
	}

	fun replaceItem(item: ItemStack): ItemStack {
		val sname = item.i18NDisplayName
		item.type = type
		item.durability = data
		val im = item.itemMeta
		im.setDisplayName("$name §b($sname)")
		im.isUnbreakable = true
		im.addItemFlags(ItemFlag.HIDE_UNBREAKABLE, ItemFlag.HIDE_ATTRIBUTES)
		item.itemMeta = im
		return item
	}

	//region static
	companion object {
		private val list: MutableMap<String, Sword> = HashMap()

		@JvmStatic
		operator fun get(id: String) = list[id]

		@JvmStatic
		fun load() {
			val main = Minigames.getInstance()
			main.logger.info("Swords...")
			val config = YamlConfiguration
				.loadConfiguration(File(Configs.getMasterPath() + "swords.yml"))
			config.getKeys(false).forEach { key: String ->
				try {
					val name = config.getString("$key.name", "")!!.colorCodes()
					var hitSound = config.getString("$key.hitsound")
					if (hitSound != null) hitSound = hitSound.colorCodes()
					val lore = config.getStringList("$key.description").colorCodes()
					val data = config.getInt("$key.data").toShort()
					list[key] = Sword(key, name, Material.IRON_SWORD, data, hitSound, lore)
				} catch (e: Exception) {
					main.logger.log(Level.WARNING, "Can't load $key sword", e)
				}
			}
		}

		@JvmStatic
		fun getSuggestions() = list.values.map { x: Sword -> "SwordPI$" + x.iD + "$0" }
	}
	//endregion
}